src/main/java/carpet/CarpetSettings.java

```diff
@@ -739,4 +741,47 @@ public static void changeSpawnSize(int size)
             category = {SURVIVAL, CLIENT}
     )
     public static boolean cleanLogs = false;
+
+    @Rule(
+            desc = "Customizable structure block limit of each axis",
+            extra = {"WARNING: Needs to be permanent for correct loading.",
+                    "Setting 'structureBlockIgnored' to air is recommended",
+                    "when saving massive structures.",
+                    "Required on client of player editing the Structure Block.",
+                    "'structureBlockOutlineDistance' may be required for",
+                    "correct rendering of long structures."},
+            options = {"48", "96", "192", "256"},
+            category = {CREATIVE, CLIENT},
+            strict = false
+    )
+    public static int structureBlockLimit = 48;
+
+    public static class StructureBlockIgnoredValidator extends Validator<String> {
+
+        @Override
+        public String validate(ServerCommandSource source, ParsedRule<String> currentRule, String newValue, String string) {
+            if (!Registry.BLOCK.getIds().contains(Identifier.tryParse(newValue))) {
+                Messenger.m(source, "Unknown block '" + newValue + "'.");
+                return null;
+            }
+            return newValue;
+        }
+    }
+    @Rule(
+            desc = "Changes the block ignored by the Structure Block",
+            options = {"minecraft:structure_void", "minecraft:air"},
```

Comment:
I think you can use non-namespaced ids as long as you are in the `minecraft` namespace, which may look better.

Reply:
Vanilla auto-completion adds the `minecraft` namespace, and that's why I did it that way.

Reply:
Well, Carpet usually removes the namespace if it equals `minecraft` (at least in Scarpet), that's why I said that. I suppose it doesn't really matter.

See, for example, `carpet.script.value.NBTSerializableValue#L102` or `carpet.script.value.ValueConversions#L176-187`

https://github.com/gnembon/fabric-carpet/blob/a0067d0d6122dcd5394eba1e76f1119773fd45b6/src/main/java/carpet/script/value/NBTSerializableValue.java#L102-L104

https://github.com/gnembon/fabric-carpet/blob/08adaaeffd3e7f18026fa083f1a8f1036090a7a5/src/main/java/carpet/script/value/ValueConversions.java#L176-L187

Reply:
well, that's scarpet - where 'minecraft:' namespace is implied. As long as it behaves similarly to other vanilla commands like `/setblock` I am fine with